meta-resin-common/classes/image_types_resin_ubi.bbclass

```diff
@@ -0,0 +1,168 @@
+        esac
+
+        # Check src type and existence
+        if [ -d "$src" ]; then
+            if ! $dst_is_dir; then
+                bbfatal "You can't copy a directory to a file. You requested to copy $src in $dst."
+            fi
+            sources="$(find $src -maxdepth 1 -type f)"
+        elif [ -f "$src" ]; then
+            sources="$src"
+        else
+            bbfatal "$src is an invalid path referenced in RESIN_BOOT_PARTITION_FILES."
+        fi
+
+        # Normalize paths
+        dst=$(realpath -ms $dst)
+        if $dst_is_dir && [ ! "$dst" = "/" ]; then
+            dst="$dst/" # realpath removes last '/' which we need to instruct mcopy that destination is a directory
+        fi
+        src=$(realpath -m $src)
+
+        for src in $sources; do
+            echo "Copying $src -> $dst ..."
+            # Create the directories parent directories in dst
+            directory="${RESIN_BOOT_DIR}"
+            for path_segment in $(echo ${dst} | sed 's|/|\n|g' | head -n -1); do
+                if [ -z "$path_segment" ]; then
+                    continue
+                fi
+                directory=$directory/$path_segment
+                mkdir $directory || true
+            done
+            cp -r ${src} ${RESIN_BOOT_DIR}/${dst}
+            if $dst_is_dir; then
+                md5sum ${src} | awk -v awkdst="$dst$(basename $src)" '{ $2=awkdst; print }' >> ${WORKDIR}/${RESIN_BOOT_FS_LABEL}.${FINGERPRINT_EXT}
+            else
+                md5sum ${src} | awk -v awkdst="$dst" '{ $2=awkdst; print }' >> ${WORKDIR}/${RESIN_BOOT_FS_LABEL}.${FINGERPRINT_EXT}
+            fi
+        done
+    done
+    # Add stamp file
+    echo "${IMAGE_NAME}-${IMAGEDATESTAMP}" > ${WORKDIR}/image-version-info
+    cp ${WORKDIR}/image-version-info ${RESIN_BOOT_DIR}
+    md5sum ${WORKDIR}/image-version-info | awk -v filepath="/image-version-info" '{ $2=filepath; print }' >> ${WORKDIR}/${RESIN_BOOT_FS_LABEL}.${FINGERPRINT_EXT}
+    # Finally add the fingerprint to boot
+    cp ${WORKDIR}/${RESIN_BOOT_FS_LABEL}.${FINGERPRINT_EXT} ${RESIN_BOOT_DIR}
+
+    init_config_json ${WORKDIR}
+
+    cp ${WORKDIR}/config.json ${RESIN_BOOT_DIR}/config.json.init
+
+	      # Split MKUBIFS_ARGS_<name> and UBINIZE_ARGS_<name>
+	      for name in ${UBIMULTIVOL_BUILD}; do
+	      	      eval local mkubifs_args=\"\$MKUBIFS_ARGS_${name}\"
+                eval local additional_ubinize_args=\"\$ADDITIONAL_UBINIZE_ARGS_${name}\"
+                eval local additional_mkubifs_args=\"\$ADDITIONAL_MKUBIFS_ARGS_${name}\"
+
+	              multiubivol_mkfs "${mkubifs_args}" "${additional_mkubifs_args}" "${additional_ubinize_args}" "${name}" "${UBI_VOLNAME}"
+	done
+        multiubivol_ubinize ${UBI_VOLNAME}
+}
+
+IMAGE_DEPENDS_multiubivol = "mtd-utils-native"
+
+# This variable is available to request which values are suitable for IMAGE_FSTYPES
+IMAGE_TYPES = " \
+    multiubivol \
```

Comment:
Hi @dlx can we get the resin specific code in this file in the image_types_resin.bbclass and remove image_types_resin_ubi.bbclass completely?
cc @michal-mazurek @agherzan